build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -108,12 +109,126 @@ service Execution {
+  //
+  // This call may not be implemented on all platforms. Its support is
+  // indicated by
+  // [ExecutionCapabilities.dynamic_execution_support][build.bazel.remote.execution.v2.ExecutionCapabilities.dynamic_execution_support].
+  //
+  // In order to execute an action, the client must first upload the best
+  // guesses for the action's file inputs (which can be zero files if
+  // prediction is unavailable), the
+  // [Command][build.bazel.remote.execution.v2.Command] to run, and the
+  // [Action][build.bazel.remote.execution.v2.Action] containing the best
+  // guess for the `Action`'s needed directory and file layout into the
+  // [ContentAddressableStorage][build.bazel.remote.execution.v2.ContentAddressableStorage].
+  // It then calls `ExecuteWithDynamicRetrieval` with an `action_digest`
+  // referring to them. The worker will run the action and eventually return
+  // the result.
+  //
+  // The `Action.input_root_digest` directory tree provides a filesystem
+  // layout that may be underspecified in terms of missing directories or
+  // unspecified file hashes. Any `Directory` entry provided in this tree
+  // must contain at least the subdirectory names and file names and sizes
+  // needed to enumerate the complete contents of the directory.
+  //
+  // A typical implementation of a server using dynamic execution will use a
+  // virtual filesystem to allow receiving callbacks about files and
+  // directories requested via operating system filesytem APIs. The server
+  // MAY allow rendering of a fully qualified, rooted filesystem by the
+  // client, by accepting in `Action.input_root_digest` a `Directory` tree
+  // that contains a filesystem root appropriate to the operating system.
+  // By convention, representation of such a filesystem utilizes a Directory
+  // with an empty string for a name. Under Linux this translates into the
+  // '/' root. Under Windows such a root contains `DirectoryNode`s that are
+  // the single-character drive letters available on the client, allowing
+  // the server to present a virtual filesystem spanning all drive roots
+  // for dynamic request by the server. There must be exactly one such
+  // empty-string root at the root of the directory tree; any use of empty
+  // strings other than the root SHOULD result in an `INVALID_ARGUMENT`
+  // error.
+  //
+  // This API is bidirectional both from the client to server and from server
+  // to client to allow the client to answer questions from the server about
+  // the contents of unspecified directories, file hashes, or to upload file
+  // content on demand, based on the observed filesystem calls of the action.
+  // This allows the server to execute actions whose inputs are incompletely
+  // specified. It also prevents execution of the action in a fire-and-forget
+  // fashion; the client must remain connected to the server for the entire
+  // execution of the action, or else the server MUST cancel the in-progress
```

Comment:
Is there a strong reason to make this a MUST rather than a SHOULD? The server might see if the action completes without further need for inputs, perhaps.

Reply:
No strong reason. Loosened the wording here to discuss the possibility of letting the server decide based on its need for dynamic information.

Reply:
Kept the indication that WaitExecution is not allowed; LMK if you disagree with it in the face of the new wording.